rpc/modules/system_test.go

```diff
@@ -8,31 +8,98 @@ import (
 
 var (
 	testRuntimeVersion = "1.2.3"
+	testRuntimeName    = "Gossamer"
+	testPeerId         = "Qmc85Ephxa3sR7xaTzTq2UpCJ4a4HWAfxxaV6TarXHWVVh"
+	peers              = []string{"QmeQeqpf3fz3CG2ckQq3CUWwUnyT2cqxJepHpjji7ehVtX", "AbCDeqpf3fz3CG2ckQq3CUWwUnyT2cqxJepHpjji7ehVtX"}
+	isSyncing          = false
 )
 
 type mockruntimeApi struct{}
+type mockP2PApi struct{}
 
+//Mock runtime API
 func (a *mockruntimeApi) Version() string {
 	return testRuntimeVersion
 }
 
+func (a *mockruntimeApi) Name() string {
+	return testRuntimeName
+}
+
+//Mock p2p API
+func (a *mockP2PApi) PeerCount() int {
+	return len(peers)
+}
+
+func (a *mockP2PApi) Peers() []string {
+	return peers
+}
+
+func (a *mockP2PApi) ShouldHavePeers() bool {
+	return (len(peers) != 0)
+}
+
+func (a *mockP2PApi) ID() string {
+	return testPeerId
+}
+
 func newMockApi() *api.Api {
 	runtimeApi := &mockruntimeApi{}
+	p2pApi := &mockP2PApi{}
 
 	return &api.Api{
-		System: api.NewSystemModule(nil, runtimeApi),
+		P2pSystem: api.NewP2PModule(p2pApi),
+		RtSystem:  api.NewRTModule(runtimeApi),
 	}
 }
 
-func TestSystemModule_Version(t *testing.T) {
+func TestSystemModule(t *testing.T) {
```

Comment:
Please move each of these to their own test. Alternatively you could implement table driven tests. Whatever is quickest pls